src/main/cpp/src/utilities.hpp

```diff
@@ -26,6 +28,57 @@
 
 namespace spark_rapids_jni {
 
+/**
+ * @brief Element accessor from a column device view.
+ * It has the same interfaces as the `numeric_scalar_accessor`.
+ * This is used to access element from column and scalar with a unified model.
+ */
+template <typename T>
+class column_accessor {
+ public:
+  column_accessor(cudf::column_device_view cdv) : _cdv(cdv) {}
+
+  __device__ bool is_null(cudf::size_type row_index) const { return _cdv.is_null(row_index); }
```

Comment:
Do not put device code in `hpp` file. Make a `cuh` file instead.

Reply:
Now is using cudf::detail::make_pair_iterator, so removed the device code in this hpp